src/azure/Model/MethodRba.cs

```diff
@@ -205,6 +207,10 @@ public override string InitializeResponseBody
 
                 sb.AppendLine(
                     "result.request_id = http_response['{0}'] unless http_response['{0}'].nil?", this.RequestIdString);
+                sb.AppendLine(
+                    "result.correlation_request_id = http_response['{0}'] unless http_response['{0}'].nil?", this.CorrelationRequestIdString);
+                sb.AppendLine(

```

Comment:
please double check that this doesn't get overwritten when setting default headers https://github.com/Azure/autorest.ruby/blob/d2eaa22768f058926f7fec8e52990416aec2507f/src/azure/Model/MethodRba.cs#L232

Reply:
Confirmed. It does not override.